sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml

```diff
@@ -26,6 +26,10 @@
 
   <packaging>jar</packaging>
 
+  <properties>
+    <beam.version>0.4.0-incubating-SNAPSHOT</beam.version>
```

Comment:
Note that this is in the `src` directory rather than an actual `pom.xml`. So far, we have not discovered a good way to propagate things from the root `pom.xml` into the source code for archetypes. I think that @kennknowles was trying some things here, but not sure if any progress was made.

I generally consider this an orthogonal issue, but am happy to discuss if not.